src/components/availability-panel/drag-drop-context/DraggableComponent.tsx

```diff
@@ -33,6 +33,14 @@ const DraggableComponent: FC<draggableProps> = ({
   title = '',
 }) => {
   const draggableIds = useContext(disableDrag);
+  const [draggableImageUrl, setDraggableImageUrl] = useState(DUMMY_PROFILE);
+  const draggableImageOnError = useCallback(() => setDraggableImageUrl(DUMMY_PROFILE), []);
```

Comment:
do we really need to use `useCallback` here for setting a state? as I don't see a reason to add setState in useCallback. Also, the dependency array is empty

Reply:
Using `useCallback`,  we are ensuring that the function is memoized and only gets redefined if the **draggableId** changes. This helps to prevent unnecessary re-renders and optimize the performance of the component. its a small improvement but @shmbajaj  said its not necessary then i will remove it.